src/App.tsx

```diff
@@ -1,61 +1,152 @@
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const clearForm = () => {
+    setTitle('');
+    setUser('');
+  };
+
+  const getMaxId = (): number => {
+    const todosIds = todos.map(todo => todo.id);
+    const largestId = Math.max(...todosIds);
+
+    return largestId;
+  };
+
+  const getIdByUsername = (userName: string): number | null => {
+    return usersFromServer.find(person => person.name === userName)?.id || null;
+  };
+
+  const getUserByUsername = (userName: string): User | null => {
+    return usersFromServer.find(
+      person => person.name === userName,
+    ) || null;
+  };
+
+  const handleAdding = () => {
+    const todoToAdd = {
+      id: getMaxId() + 1,
+      userId: getIdByUsername(user),
+      title,
+      completed: false,
+      user: getUserByUsername(user),
+    };
+
+    setTodos((prev) => ([...prev, todoToAdd]));
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title === '') {
+      setTitleError(true);
+    }
+
+    if (user === '') {
+      setUserError(true);
+    }
+
+    if (title && user) {
+      handleAdding();
+      clearForm();
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            <input
+              type="text"
+              name="title"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={(event) => {
+                setTitle(event.target.value);
+                setTitleError(false);
+              }}
+            />
+          </label>
+
+          {titleError && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            User:
+            <select
+              name="user"
+              data-cy="userSelect"
+              value={user}
+              onChange={(event) => {
+                setUser(event.target.value);
+                setUserError(false);
+              }}
+            >
+              <option value="" disabled>
+                Choose a user
+              </option>
+
+              {usersFromServer.map(person => (
+                <option
+                  key={person.id}
+                  value={person.name}
+                >
+                  {person.name}
+                </option>
+              ))}
```

Comment:
```suggestion
              {usersFromServer.map(user => (
                <option
                  key={user.id}
                  value={user.name}
                >
                  {user.name}
                </option>
              ))}
```

Reply:
If I use a 'user' variable, the error ''user' is already declared in the upper scope' appears.